### PR TITLE
Warn about deprecation at runtime

### PR DESCRIPTION
**Pull request by dwt**
This is an enhancement of the pull request #14. Basically it is meant to give developers who are using this package already a chance to discover its status.

I am not sure this is the right / best way to document deprecation status in Zope, but a quick grep didn't show an obvious more advanced pattern - so this is where I started.

Suggestions for improvement are welcomed.

### CODE REVIEW

File: src/tempstorage/TemporaryStorage.py [hunk 1]
```diff
@@ -84,6 +85,14 @@ def __init__(self, name='TemporaryStorage'):
 
         _conflict_cache_maxage -- age at whic conflict cache items are GC'ed
         """
+        deprecation_warning = """\
+DEPRECATED: Usage of the package tempstorage is deprecated, as it is known to randomly lose data.
+Especially on zope 4. For details see https://github.com/zopefoundation/tempstorage/issues/8
```
Comment (icemac) [inline src/tempstorage/TemporaryStorage.py]:
```suggestion
Especially on Zope 4. For details see https://github.com/zopefoundation/tempstorage/issues/8
```

File: src/tempstorage/TemporaryStorage.py [hunk 2]
```diff
@@ -84,6 +85,14 @@ def __init__(self, name='TemporaryStorage'):
...
+and https://github.com/zopefoundation/tempstorage
+"""
+        LOG.warning(deprecation_warning)
+        warnings.warn(deprecation_warning, DeprecationWarning)
```
Comment (icemac) [inline src/tempstorage/TemporaryStorage.py]: Do we need both logging and warnings?

Comment (dwt) [reply src/tempstorage/TemporaryStorage.py]: The warning is visible in the log by default, while the warning is not, but is on the other hand searchable with tooling. Thus both - but I am not familiar with how this is handled in other parts of zope, so I am open to suggestions.

Comment (icemac) [reply src/tempstorage/TemporaryStorage.py]: There are many ways, e. g. [zope.deprecation](https://pypi.org/project/zope.deprecation/).

Comment (dwt) [reply src/tempstorage/TemporaryStorage.py]: That module seems to only call warnings.warn(..., DeprecationWarning) while being much more complex and adding a dependency just for this.

Knowing that, I would prefer to have this simple solution that also guarantees that people get the log entry.

Comment (dwt) [reply src/tempstorage/TemporaryStorage.py]: @icemac Are you ok with that? If so, please merge or indicate consent so I can do that.

**Comment by icemac** [comment]
Released as 5.1 to PyPI, see https://pypi.org/project/tempstorage/5.1/

**Pull request merged**